packages/jest-matchers/src/index.js

```diff
@@ -78,9 +84,27 @@ const addMatchers = (matchersObj: MatchersObject): void => {
   Object.assign(global[GLOBAL_MATCHERS_OBJECT_SYMBOL], matchersObj);
 };
 
+const _validateResult = result => {
```

Comment:
There aren't any unnecessary bind calls because this is run in node 4 without compiling arrow functions to functions. Also babel doesn't bind if there is no this access iirc.